op2/src/mpi/op_mpi_part_core.cpp

```diff
@@ -68,6 +68,14 @@ typedef int idx_t;
 #endif
 #endif
 
+// kaHIP header
+#ifdef HAVE_KAHIP
+#include <parhip_interface.h>
+
+typedef idxtype idx_t;
```

Comment:
I am getting a compile time error with gcc 9 if I also have ParMetis installed:

> In file included from src/mpi/op_mpi_part_core.cpp:73:
/home/shared/software/kahip-ompi401/include/parhip_interface.h:15:28: error: conflicting declaration ‘typedef long long unsigned int idxtype’
   15 | typedef unsigned long long idxtype;
      |                            ^~~~~~~
src/mpi/op_mpi_part_core.cpp:65:15: note: previous declaration as ‘typedef idx_t idxtype’
   65 | typedef idx_t idxtype;

Reply:
I ran into the same issue yesterday, looks like ParMETIS defines `idx_t` while ParHIP defines `idxtype`, which is distinct. As long as only one of the two are installed, we can use `typedef` to set the type as necessary but with both installed we're forced to choose between the types at runtime. 

The interface for the two k-way partition functions are very similar apart from this type issue so I'm reluctant to define another `op_partition_kway` just for the ParHIP version. But I've yet to find an elegant solution.

Reply:
That's a tough one... My first idea, while being wasteful, but at least compact, would be to pass the parmetis-style arrays to a function, and allocate new ones with kahip types, convert them, then call kahip. Which is indeed ugly! But the fundamental issue is that even if both use 64 bits, kahip uses unsigned, whereas parmetis signed...